erpnext/accounts/doctype/shipping_rule/shipping_rule.py

```diff
@@ -113,7 +113,34 @@ def add_shipping_rule_to_tax_table(self, doc, shipping_amount):
 		else:
 			shipping_charge["tax_amount"] = shipping_amount
 			shipping_charge["description"] = self.label
-			doc.append("taxes", shipping_charge)
+
+			# Check if already tax is present & add new shipping taxes
+			if doc.get("taxes"):
+				taxes_and_charges = []
+				new_tax = doc.get("taxes")
+				for d in doc.get("taxes"):
+					taxes_and_charges.append(d)
+
+				doc.set("taxes", [])
+				doc.append("taxes", shipping_charge)
+
+				for tax in taxes_and_charges:
+					doc.append("taxes", tax)
```

Comment:
```suggestion
				doc.set("taxes", shipping_charge + taxes_and_charges)
```

Reply:
Shipping charges & taxes charges are in a different format(one is a list and the other is dict) so we cannot append with the + sign, also for loop is used for if taxes & charges are having more than one record.